src/main/java/blusunrize/immersiveengineering/common/blocks/metal/SawmillTileEntity.java

```diff
@@ -0,0 +1,596 @@
+		super.readCustomNBT(nbt, descPacket);
+
+		sawblade = ItemStack.read(nbt.getCompound("sawblade"));
+
+		ListNBT processNBT = nbt.getList("sawmillQueue", 10);
+		sawmillProcessQueue.clear();
+		for(int i = 0; i < processNBT.size(); i++)
+		{
+			CompoundNBT tag = processNBT.getCompound(i);
+			SawmillProcess process = SawmillProcess.readFromNBT(tag);
+			sawmillProcessQueue.add(process);
+		}
+	}
+
+	@Override
+	public void writeCustomNBT(CompoundNBT nbt, boolean descPacket)
+	{
+		super.writeCustomNBT(nbt, descPacket);
+		if(!this.sawblade.isEmpty())
+			nbt.put("sawblade", this.sawblade.write(new CompoundNBT()));
+		ListNBT processNBT = new ListNBT();
+		for(SawmillProcess process : this.sawmillProcessQueue)
+			processNBT.add(process.writeToNBT());
+		nbt.put("sawmillQueue", processNBT);
+	}
+
+	@Override
+	public void receiveMessageFromClient(CompoundNBT message)
+	{
+	}
+
+	private CapabilityReference<IItemHandler> outputCap = CapabilityReference.forTileEntity(this, () -> {
+		Direction outDir = getIsMirrored()?getFacing().rotateYCCW(): getFacing().rotateY();
+		return new DirectionalBlockPos(getBlockPosForPos(new BlockPos(4, 1, 1)).offset(outDir), outDir.getOpposite());
+	}, CapabilityItemHandler.ITEM_HANDLER_CAPABILITY);
+
+	private CapabilityReference<IItemHandler> secondaryOutputCap = CapabilityReference.forTileEntity(this, () -> {
+		Direction shiftDir = getIsMirrored()?getFacing().rotateYCCW(): getFacing().rotateY();
+		return new DirectionalBlockPos(getBlockPosForPos(new BlockPos(3, 1, 2)).offset(shiftDir), getFacing().getOpposite());
+	}, CapabilityItemHandler.ITEM_HANDLER_CAPABILITY);
+
+	@Override
+	public void tick()
+	{
+		super.tick();
+		if(isDummy()||isRSDisabled())
+			return;
+		if(world.isRemote&&shouldRenderAsActive())
+		{
+			animation_bladeRotation += 36f;
+			animation_bladeRotation %= 360f;
+		}
+		if(world.isRemote)
+			return;
+
+		tickedProcesses = 0;
+		int max = getMaxProcessPerTick();
+		int i = 0;
+		Iterator<SawmillProcess> processIterator = sawmillProcessQueue.iterator();
+		tickedProcesses = 0;
+		Set<ItemStack> secondaries = new HashSet<>();
+		while(processIterator.hasNext()&&i++ < max)
+		{
+			SawmillProcess process = processIterator.next();
+			if(process.processStep(this, secondaries))
+				tickedProcesses++;
+			if(process.processFinished)
+			{
+				doProcessOutput(process.getCurrentStack(!this.sawblade.isEmpty()).copy());
+				processIterator.remove();
+				// todo: make the damage configurable
+				if(this.sawblade.attemptDamageItem(5, Utils.RAND, null))
+				{
+					this.sawblade = ItemStack.EMPTY;
+					this.updateMasterBlock(null, true);
+				}
+			}
+		}
+		for(ItemStack output : secondaries)
+			doSecondaryOutput(output.copy());
+	}
+
+	@Override
+	public boolean interact(Direction side, PlayerEntity player, Hand hand, ItemStack heldItem, float hitX, float hitY, float hitZ)
+	{
+		SawmillTileEntity master = master();
+		if(master!=null)
+			if(player.isSneaking()&&!master.sawblade.isEmpty())
```

Comment:
I've said this before, I don't really like `if(a) if(b) … else` without braces on the outer if. Could you add those?